plutus-ir/src/Language/PlutusIR/Compiler/Datatype.hs

```diff
@@ -277,33 +313,37 @@ mkConstructor r pf (Datatype _ tn tvs _ constrs) constr = do
 -- See note [Scott encoding of datatypes]
 -- | Make the destructor for a 'Datatype'.
 -- @
---     mkDestructor List = /\(a :: *) . \(x : List a) . unwrap x
+--     mkDestructor <definition of List> List =
+--        /\(a :: *) -> \(x : (<definition of List> a)) -> unwrap x =
+--        /\(a :: *) -> \(x : (fix list . \(a :: *) -> forall (r :: *) . r -> (a -> List a -> r) -> r) a) -> unwrap x
```

Comment:
Lowercase `list` in `fix list`.

It's hard to read (to me) things like

```
\(x : (<definition of List> a)) -> unwrap x =
```

I'd prefer

```
= /\(a :: *) -> \(x : (<definition of List> a)) -> unwrap x
= /\(a :: *) -> \(x : (fix list . \(a :: *) -> forall (r :: *) . ...
```

Reply:
Sure, I can put the equalities at the beginning.